src/pages/academy/xpCalculation/XpCalculation.tsx

```diff
@@ -0,0 +1,249 @@
+import 'ag-grid-community/dist/styles/ag-grid.css';
+import 'ag-grid-community/dist/styles/ag-theme-balham.css';
+
+import { Button, Divider, NonIdealState, Spinner, SpinnerSize } from '@blueprintjs/core';
+import { IconNames } from '@blueprintjs/icons';
+import { GridApi, GridReadyEvent } from 'ag-grid-community';
+import { AgGridReact } from 'ag-grid-react';
+import { useEffect, useState } from 'react';
+import ContentDisplay from 'src/commons/ContentDisplay';
+
+export type StateProps = {
+  allUserXp: string[][] | undefined;
+};
+
+export type DispatchProps = {
+  handleAllUserXpFetch: () => void;
+};
+
+type XpCalculationProps = DispatchProps & StateProps;
+
+type RowData = {
+  name: string;
+  'NUS Net ID': string;
+  'Assessment Xp': string;
+  'Achievement Xp': string;
+};
+
+type PageState = {
+  currPage: number;
+  maxPages: number;
+  rowCountString: string;
+  isBackDisabled: boolean;
+  isForwardDisabled: boolean;
+};
+
+const XpCalculation = ({ allUserXp, handleAllUserXpFetch }: XpCalculationProps) => {
```

Comment:
I think this typing might be better:

```suggestion
const XpCalculation: React.FC<XpCalculationProps> = ({ allUserXp, handleAllUserXpFetch }) => {
```